providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/grpc/GrpcConnector.java

```diff
@@ -80,12 +82,13 @@ public void initialize() throws RuntimeException {
             }
         } finally {
             // try in background to open the event stream
-            this.startEventStream();
+            new Thread(this::observeEventStream).start();
```

Comment:
> Perhaps can use a single threaded (or other) executor, and then shut it down gracefully at shutdown() method ?

Ya, I generally like to use single thread executors instead of threads directly.

Reply:
Unfortunately, we need some sort of a concurrent mechanism to execute the connection retry logic [^1]. This is to support, busy wait (sleep) and the loop for retry counter.

However, I agree with the point of graceful shutdown (thread termination on shutdown invocation). I will work on this improvement 

[^1]: https://github.com/Kavindu-Dodan/java-sdk-contrib/blob/feat/refactor-flagd/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/grpc/GrpcConnector.java#L115-L144